events/source.go

```diff
@@ -300,30 +297,19 @@ func (q *queryBlockNotifier) BlockHeights(ctx context.Context) (<-chan int64, <-
 	return blocks, errChan
 }
 
-func (q *queryBlockNotifier) connectNewClient() error {
-	var err error
-	q.client, err = q.getClient()
-	return err
-}
-
 func (q *queryBlockNotifier) latestFromSyncStatus(ctx context.Context) (int64, error) {
-	i := 0
-
-	tryQuery := func() (int64, error) {
+	backOff := utils.LinearBackOff(q.backOff)
+	for i := 0; i <= q.retries; i++ {
 		if q.client == nil || i > 0 {
-			if err := q.connectNewClient(); err != nil {
+			var err error
+			if q.client, err = q.clientFactory.Create(); err != nil {
 				return 0, err
 			}
 		}
```

Comment:
This breaks the fault tolerance. Now if the new client connection fails, the retry loop will be ignored.

Reply:
good catch